runtime/syntax/man.vim

```diff
@@ -1,67 +1,39 @@
-" Vim syntax file
-" Language:	Man page
-" Maintainer:	SungHyun Nam <goweol@gmail.com>
-" Previous Maintainer:	Gautam H. Mudunuri <gmudunur@informatica.com>
-" Version Info:
-" Last Change:	2008 Sep 17
-
-" Additional highlighting by Johannes Tanzler <johannes.tanzler@aon.at>:
-"	* manSubHeading
-"	* manSynopsis (only for sections 2 and 3)
-
-" For version 5.x: Clear all syntax items
-" For version 6.x: Quit when a syntax file was already loaded
-if version < 600
-  syntax clear
-elseif exists("b:current_syntax")
+if exists("b:current_syntax")
   finish
 endif
 
-" Get the CTRL-H syntax to handle backspaced text
-if version >= 600
-  runtime! syntax/ctrlh.vim
-else
-  source <sfile>:p:h/ctrlh.vim
-endif
-
-syn case ignore
-syn match  manReference       "\f\+([1-9][a-z]\=)"
-syn match  manTitle	      "^\f\+([0-9]\+[a-z]\=).*"
-syn match  manSectionHeading  "^[a-z][a-z ]*[a-z]$"
-syn match  manSubHeading      "^\s\{3\}[a-z][a-z ]*[a-z]$"
-syn match  manOptionDesc      "^\s*[+-][a-z0-9]\S*"
-syn match  manLongOptionDesc  "^\s*--[a-z0-9-]\S*"
-" syn match  manHistory		"^[a-z].*last change.*$"
-
-if getline(1) =~ '^[a-zA-Z_]\+([23])'
-  syntax include @cCode <sfile>:p:h/c.vim
-  syn match manCFuncDefinition  display "\<\h\w*\>\s*("me=e-1 contained
-  syn region manSynopsis start="^SYNOPSIS"hs=s+8 end="^\u\+\s*$"me=e-12 keepend contains=manSectionHeading,@cCode,manCFuncDefinition
-endif
-
-
-" Define the default highlighting.
-" For version 5.7 and earlier: only when not done already
-" For version 5.8 and later: only when an item doesn't have highlighting yet
-if version >= 508 || !exists("did_man_syn_inits")
-  if version < 508
-    let did_man_syn_inits = 1
-    command -nargs=+ HiLink hi link <args>
-  else
-    command -nargs=+ HiLink hi def link <args>
+syntax case  ignore
+syntax match manReference       "\f\+(\%([0-8][a-z]\=\|n\))"
+syntax match manTitle           "^\%1l\S\+\%((\%([0-8][a-z]\=\|n\))\)\=.*$"
+syntax match manSubHeading      "^\s\{3\}\%(\S.*\)\=\S$"
+syntax match manOptionDesc      "^\s\+[+-][a-z0-9]\S*"
+syntax match manLongOptionDesc  "^\s\+--[a-z0-9]\S*"
+" prevent manSectionHeading from matching last line
+execute 'syntax match manSectionHeading  "^\%(\%>1l\%<'.line('$').'l\)\%(\S.*\)\=\S$"'
+
+highlight default link manTitle          Title
+highlight default link manSectionHeading Statement
+highlight default link manOptionDesc     Constant
+highlight default link manLongOptionDesc Constant
+highlight default link manReference      PreProc
+highlight default link manSubHeading     Function
+
+if getline(1) =~# '^\f\+([23][a-zA-Z]\=)'
+  syntax include @cCode $VIMRUNTIME/syntax/c.vim
+  " skip first heading
+  keepjumps call search('^\%(\S.*\)\=\S$')
+  keepjumps call search('^\%(\S.*\)\=\S$')
+  " get second heading, aka the synopsis heading
+  let s:l = escape(getline('.'), '\')
+  " why bsd...
+  if s:l ==# 'LIBRARY'
```

Comment:
For some reason FreeBSD has a extra section between the NAME and SYNOPSIS which means I can't just always select the second section for the c highlighting...

https://www.freebsd.org/cgi/man.cgi?query=printf&apropos=0&sektion=3&manpath=FreeBSD+10.2-RELEASE&arch=default&format=html

I've added a special case, but we all hate special cases. I was thinking of a variable like `g:man_synopsis` , which contains a regex used to match the synopsis. So by default it would probably be `^SYNOPSIS$` but you could change it to add in other languages yourself.

@justinmk what do you think?